features/application/MainNavigation.feature

```diff
@@ -83,8 +83,16 @@ Feature: index
       | policies               | 100    | 99+ |
 
   Scenario Outline: Visiting the "<Title>" page
+    # Given the URL "/mesh-insights/default" responds with
+    #   """
+    #   body:
+    #     items:
+    #     - policies:
+    #         CircuitBreaker:
+    #           total: 1
+    #   """
```

Comment:
If you change this to:

```yaml
body:
  policies:
    CircuitBreaker:
      total: 0
    FaultInjection:
      total: 1
```

And then obvs:

```gherkin
      | $main-nav .nav-item-policies a               | FaultInjection |
```

Then this works fine (I chose to not choose the naturally first policy type)

There's a swallowed error pretty deep in the application that made this not obvious. I think we should make an effort to stop trying to cover up when things go wrong with the backend/binary. The GUI should act as a thin client, when things go wrong, as a user, I want to know about it, at the very least in the javascript error console.

Just in case to be clear here, things haven't gone wrong with the backend, but the above commented mocks here mocked out our mock backend to make it so things had gone wrong. Hence the test wouldn't pass (which is actually a good thing). Just it was harder than it should be to figure out what was wrong as the error was swallowed and we tried to cover up the problem in the frontend instead of just acting like a thin client and saying "hey the backend gave us something we weren't expecting"

I might add something to next weeks GUI sync to chat about this.

Reply:
P.S. to be fair we probably shouldn't make these changes here as they have nothing to do with the PR. If it was me I'd remove all this commenting/changes here and draft a new PR to specifically address this bit. But up to you 🤷